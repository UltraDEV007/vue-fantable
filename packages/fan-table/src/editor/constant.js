// instance methods
export const INSTANCE_METHODS = {
  TEXTAREA_ADD_NEW_LINE: 'textareaAddNewLine',
  TEXTAREA_SELECT: 'textareaSelect',
}
