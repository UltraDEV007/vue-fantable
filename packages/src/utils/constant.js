// key codes
export const KEY_CODES = {
  BACK_SPACE: 8,
  TAB: 9,
  ENTER: 13,
  SHIFT: 16,
  SPACE: 32,

  ARROW_LEFT: 37,
  ARROW_UP: 38,
  ARROW_RIGHT: 39,
  ARROW_DOWN: 40,

  DELETE: 46,

  F2: 113,
}

// icon name
export const ICON_NAMES = {
  FILTER: 'filter',

  DOUBLE_RIGHT_ARROW: 'double-right-arrow',
  DOUBLE_LEFT_ARROW: 'double-left-arrow',

  TOP_ARROW: 'top-arrow',
  RIGHT_ARROW: 'right-arrow',
  BOTTOM_ARROW: 'bottom-arrow',
  LEFT_ARROW: 'left-arrow',

  SORT_TOP_ARROW: 'sort-top-arrow',
  SORT_BOTTOM_ARROW: 'sort-bottom-arrow',

  SEARCH: 'search',
}

// mouse event key type
export const MOUSE_EVENT_CLICK_TYPE = {
  LEFT_MOUSE: 1,
  MIDDLE_MOUSE: 2,
  RIGHT_MOUSE: 2,
}
