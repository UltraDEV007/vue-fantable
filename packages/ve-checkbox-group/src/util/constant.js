// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
