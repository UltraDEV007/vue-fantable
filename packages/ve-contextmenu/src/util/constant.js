// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
