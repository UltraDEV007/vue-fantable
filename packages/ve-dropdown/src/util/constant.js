// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
