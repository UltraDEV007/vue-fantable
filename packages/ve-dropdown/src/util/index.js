/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
