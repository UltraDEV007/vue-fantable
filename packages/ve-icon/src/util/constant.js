// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
