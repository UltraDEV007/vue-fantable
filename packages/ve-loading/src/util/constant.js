// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
