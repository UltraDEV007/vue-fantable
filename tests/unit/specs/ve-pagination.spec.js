import { mount } from "@vue/test-utils";
import vePagination from "@/ve-pagination";
import { later } from "../util";
import { KEY_CODES } from "../constant";

describe("vePagination", () => {
    it("render by different props", () => {
        const wrapper = mount({
            render() {
                return (
                    <div>
                        <vePagination total={600} />
                        <vePagination total={600} pageIndex={2} />
                        <vePagination total={600} pageIndex={30} />
                        <vePagination
                            total={600}
                            layout={[
                                "total",
                                "prev",
                                "next",
                                "sizer",
                                "jumper",
                            ]}
                        />
                        <vePagination
                            total={600}
                            layout={[
                                "total",
                                "sizer",
                                "prev",
                                "pager",
                                "next",
                                "jumper",
                            ]}
                        />
                    </div>
                );
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("layout prop", () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 300,
                layout: ["total", "prev", "next", "sizer", "jumper"],
            },
        });

        // except pager
        expect(wrapper.find(".ve-pagination-pager").exists()).toBe(false);
    });

    it("total prop", () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 300,
            },
        });
        expect(wrapper.find(".ve-pagination-total").text()).toContain("300");
    });

    it("pageIndex prop", () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
                pageIndex: 5,
            },
        });

        expect(wrapper.find(".ve-pagination-li-active").text()).toBe("5");
    });

    it("pageSizeOption prop", () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
                pageSizeOption: [5, 10, 15],
                pageSize: 15,
            },
        });

        const pageSizeOptionEl = wrapper.findAll(".ve-dropdown-items-li");

        expect(pageSizeOptionEl.at(0).text()).toContain("5");
        expect(pageSizeOptionEl.at(1).text()).toContain("10");
        expect(pageSizeOptionEl.at(2).text()).toContain("15");
    });

    it("pagingCount prop", () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
                pagingCount: 7,
                pageIndex: 10,
            },
        });

        expect(
            wrapper.findAll(".ve-pagination-pager .ve-pagination-li").length,
        ).toBe(11);
    });

    it("pageSize prop", () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
                pageSizeOption: [5, 10, 15],
                pageSize: 15,
            },
        });

        expect(wrapper.find(".ve-dropdown-items-li.active").text()).toContain(
            "15",
        );
    });

    it("pageIndex change", async () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
            },
        });

        await wrapper.setProps({ pageIndex: 2 });

        expect(
            wrapper.find(".ve-pagination-li-active.ve-pagination-li").text(),
        ).toBe("2");
    });

    it("page number btn click operation", async () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
            },
        });

        wrapper
            .findAll(".ve-pagination-pager .ve-pagination-li")
            .at(5)
            .trigger("click");

        await later();

        expect(wrapper.emitted("on-page-number-change").length).toEqual(1);
        expect(wrapper.emitted("on-page-number-change")[0]).toEqual([6]);

        expect(
            wrapper.find(".ve-pagination-li-active.ve-pagination-li").text(),
        ).toBe("6");

        expect(wrapper.find(".ve-pagination-goto-input").element.value).toBe(
            "6",
        );
    });

    it("next page number btn click operation", async () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
            },
        });

        wrapper.find(".ve-pagination-next").trigger("click");

        await later();

        expect(wrapper.emitted("on-page-number-change").length).toEqual(1);
        expect(wrapper.emitted("on-page-number-change")[0]).toEqual([2]);

        expect(
            wrapper.find(".ve-pagination-li-active.ve-pagination-li").text(),
        ).toBe("2");
    });

    it("pre page number btn click operation", async () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
                pageIndex: 10,
            },
        });

        wrapper.find(".ve-pagination-prev").trigger("click");

        await later();

        expect(wrapper.emitted("on-page-number-change").length).toEqual(1);
        expect(wrapper.emitted("on-page-number-change")[0]).toEqual([9]);

        expect(
            wrapper.find(".ve-pagination-li-active.ve-pagination-li").text(),
        ).toBe("9");
    });

    it("next5 click operation", async () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
            },
        });

        wrapper.find(".ve-pagination-jump-next").trigger("click");

        await later();

        expect(wrapper.emitted("on-page-number-change").length).toEqual(1);
        expect(wrapper.emitted("on-page-number-change")[0]).toEqual([6]);

        expect(
            wrapper.find(".ve-pagination-li-active.ve-pagination-li").text(),
        ).toBe("6");
    });

    it("prev5 click operation", async () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
                pageIndex: 10,
            },
        });

        wrapper.find(".ve-pagination-jump-prev").trigger("click");

        await later();

        expect(wrapper.emitted("on-page-number-change").length).toEqual(1);
        expect(wrapper.emitted("on-page-number-change")[0]).toEqual([5]);

        expect(
            wrapper.find(".ve-pagination-li-active.ve-pagination-li").text(),
        ).toBe("5");
    });

    it("on-page-size-change emit", () => {
        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
                pageSizeOption: [5, 10, 15],
                pageSize: 5,
            },
        });

        wrapper.findAll(".ve-dropdown-items-li").at(1).trigger("click");

        expect(wrapper.emitted("on-page-size-change").length).toEqual(1);
        expect(wrapper.emitted("on-page-size-change")[0]).toEqual([10]);

        expect(
            wrapper.find(".ve-pagination-li-active.ve-pagination-li").text(),
        ).toBe("1");
    });

    it("enter keyboard event ", async () => {
        const mockFn = jest.fn();

        const wrapper = mount(vePagination, {
            propsData: {
                total: 600,
                pageSize: 5,
            },
        });

        const textInput = wrapper.find(".ve-pagination-goto-input");
        await textInput.setValue(2);

        expect(textInput.element.value).toBe("2");

        textInput.element.addEventListener("keyup", mockFn);
        textInput.element.dispatchEvent(
            new KeyboardEvent("keyup", { keyCode: KEY_CODES.ENTER }),
        );
        /* textInput.trigger("keyup", {
            keyCode: 13
        }); */

        await later();

        expect(mockFn).toBeCalled();

        expect(
            wrapper.find(".ve-pagination-li-active.ve-pagination-li").text(),
        ).toBe("2");

        expect(wrapper.emitted("on-page-number-change").length).toEqual(1);
        expect(wrapper.emitted("on-page-number-change")[0]).toEqual([2]);
    });
});
