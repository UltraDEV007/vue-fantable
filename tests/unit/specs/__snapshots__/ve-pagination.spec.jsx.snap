// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vePagination > render by different props 1`] = `
"<div>
  <ul class="ve-pagination"></ul>
  <ul class="ve-pagination"></ul>
  <ul class="ve-pagination"></ul>
  <ul class="ve-pagination"></ul>
  <ul class="ve-pagination"></ul>
</div>"
`;
